src/components/Account/Account.tsx

```diff
@@ -48,24 +50,39 @@ const Account = ({ panelClassName, panelHeaderClassName, canUpdateEmail = true }
   const exportData = useMutation(exportAccountData);
   const [isAlertVisible, setIsAlertVisible] = useState(false);
   const exportDataMessage = exportData.isSuccess ? t('account.export_data_success') : t('account.export_data_error');
+  const { config } = useConfigStore();
+  const accountService = useService(({ accountService }) => accountService);
 
   useEffect(() => {
     if (exportData.isSuccess || exportData.isError) {
       setIsAlertVisible(true);
     }
   }, [exportData.isSuccess, exportData.isError]);
 
-  const { customer, customerConsents, publisherConsents, canChangePasswordWithOldPassword, canExportAccountData, canDeleteAccount } = useAccountStore(
-    ({ user, customerConsents, publisherConsents, canChangePasswordWithOldPassword, canExportAccountData, canDeleteAccount }) => ({
-      customer: user,
-      customerConsents,
-      publisherConsents,
-      canChangePasswordWithOldPassword,
-      canExportAccountData,
-      canDeleteAccount,
-    }),
-    shallow,
-  );
+  const { customer, fetchOnVisit, customerConsents, publisherConsents, canChangePasswordWithOldPassword, canExportAccountData, canDeleteAccount } =
+    useAccountStore(
+      ({ user, fetchOnVisit, customerConsents, publisherConsents, canChangePasswordWithOldPassword, canExportAccountData, canDeleteAccount }) => ({
+        customer: user,
+        fetchOnVisit,
+        customerConsents,
+        publisherConsents,
+        canChangePasswordWithOldPassword,
+        canExportAccountData,
+        canDeleteAccount,
+      }),
+      shallow,
+    );
+
+  // update account with fresh data when visited
+  useEffect(() => {
+    const getUserInfo = async () => {
+      if (fetchOnVisit) {
+        const { user } = await accountService.getUser({ config, shouldFetchData: true });
+        useAccountStore.setState({ user, fetchOnVisit: false });
+      }
+    };
+    getUserInfo();
+  }, [customer, fetchOnVisit, accountService, config]);
```

Comment:
I can update this using useQuery, but we still need the fetchOnVisit (service level flag). We don't want this with cleeng. 
And also it will require creating new controller method, only for calling the getUser and updating the useAccountStore.. We do have user connected methods in the controller but they also do bunch of other calls we don't need, and I don't think it's a good idea to add additional checks there.

Reply:
updated